### ISSUE TITLE
Quick Start guide wrong, twice

### ISSUE THREAD

**Issue by Bikkelronnie** [zendframework/zend-navigation#43]
The quick start guide is wrong. First (small) issue, it has a very obvious syntax error https://docs.zendframework.com/zend-navigation/quick-start/. Look at the `array(` part in the first code block.

The second issue is worse, and it is plain wrong and its one of the weirdest parts of the navigation module in my opinion. It says to use `<?= $this->navigation('default')->menu() ?>`, but you will get an error then. You have to use `<?= $this->navigation('navigation')->menu() ?>` instead.

**Comment by froschdesign** [comment]
> The second issue is worse, and it is plain wrong…

Yes it's worse, but not the documentation is wrong – the code is wrong!

```php
<?= $this->navigation('navigation')->menu() ?>
```

Call the navigation helper with a parameter "navigation"?! Really strange! 😕

**Comment by novrm** [comment]
Also wrong.
<code><?= $this->navigation('Zend\Navigation\Default')->menu() ?></code>
Must be:
<code><?= $this->navigation('Zend\\\\Navigation\\\\Default')->menu() ?></code>

**Comment by weierophinney** [comment]
Single backslashes are fine here, as the characters following would not
create escape sequences. (Generally, only lowercase characters are used to
define escape sequences.)

**Comment by novrm** [comment]
Ok, **weierophinney**. Look [here](https://github.com/zendframework/zend-navigation/blob/master/src/Service/NavigationAbstractServiceFactory.php#L37).
The `SERVICE_PREFIX = 'Zend\\Navigation\\'` have two backslashes.
If apply in code [NavigationAbstractServiceFactory](https://github.com/zendframework/zend-navigation/blob/master/doc/book/quick-start.md#container-names-have-a-prefix) - you must have two backslashe in name of navigation.

**Comment by froschdesign** [comment]
@novrm 
> you must have two backslashe in name of navigation

The same here: changing the documentation for strange parameters? No, the code is wrong!

**Comment by froschdesign** [comment]
@novrm 
I can not reproduce the your problem. A have tested `zend-mvc` with version 3.0.4 and `zend-navigation` with 2.8.1.
Which versions do you use?

**Comment by novrm** [comment]
@froschdesign
Code below not work in my blog:
```twig
{% block layoutBodyHeaderBottomNavbar %}
<div class="navbar navbar-inverse navbar-default">
    <div class="container-fluid">
        {% block layoutBodyHeaderBottomNavbarHeader %}{{ parent() }}{% endblock layoutBodyHeaderBottomNavbarHeader %}
        <div class="collapse navbar-collapse">
        {% block layoutBodyHeaderBottomNavbarCollapseNav %}
            {{ navigation('Zend\Navigation\layoutBodyHeaderBottomNavbarCollapseNav').menu().escapeLabels(false)|raw }}
        {% endblock layoutBodyHeaderBottomNavbarCollapseNav %}
        </div>
    </div>
</div>
{% endblock layoutBodyHeaderBottomNavbar %}
```

> Fatal error: Uncaught exception 'Zend\ServiceManager\Exception\ServiceNotFoundException' with message 'Unable to resolve service "ZendNavigationlayoutBodyHeaderBottomNavbarCollapseNav" to a factory; are you certain you provided it during configuration?' in F:\xampp\htdocs\blog\vendor\zendframework\zend-servicemanager\src\ServiceManager.php:670 Stack trace: #0 F:\xampp\htdocs\blog\vendor\zendframework\zend-servicemanager\src\ServiceManager.php(746): Zend\ServiceManager\ServiceManager->getFactory('ZendNavigationl...') #1 F:\xampp\htdocs\blog\vendor\zendframework\zend-servicemanager\src\ServiceManager.php(195): Zend\ServiceManager\ServiceManager->doCreate('ZendNavigationl...') #2 F:\xampp\htdocs\blog\vendor\zendframework\zend-view\src\Helper\Navigation\AbstractHelper.php(257): Zend\ServiceManager\ServiceManager->get('ZendNavigationl...') #3 F:\xampp\htdocs\blog\vendor\zendframework\zend-view\src\Helper\Navigation\AbstractHelper.php(529): Zend\View\Helper\Navigation\AbstractHelper->parseContainer('ZendNavigationl...') #4 F:\x in F:\xampp\htdocs\blog\vendor\twig\twig\lib\Twig\Template.php on line 222

**Comment by froschdesign** [comment]
@novrm 
Please have a look at the stack trace and you will see, the backslashes are missing:

> `Zend\View\Helper\Navigation\AbstractHelper->parseContainer('ZendNavigationl...')`

Change your Twig template:

```php
{{ navigation('Zend\\Navigation\\layoutBodyHeaderBottomNavbarCollapseNav').menu().escapeLabels(false)|raw }}
```

**This is not a Zend Framework problem and not needed with `zend-view`. The double backslashes are only used in a Twig template!**

**Comment by novrm** [comment]
@froschdesign 
You are right. I know as write correct code with Zend Navigation, but...
If follow to Zend Navigation [guide - we are find not correct code](https://github.com/zendframework/zend-navigation/blob/master/doc/book/quick-start.md#container-names-have-a-prefix):
```
There is one important point to know when using NavigationAbstractServiceFactory:
The name of the service in your view must start with Zend\Navigation\ followed by the
name of the configuration key.
This helps ensure that no naming collisions occur with other services.
The following example demonstrates rendering the navigation menus for the named
default, special and sitemap containers.
```
```php
<!-- in your layout -->
<!-- ... -->

<body>
    <?= $this->navigation('Zend\Navigation\Default')->menu() ?>

    <?= $this->navigation('Zend\Navigation\Special')->menu() ?>

    <?= $this->navigation('Zend\Navigation\Sitemap')->menu() ?>
</body>
<!-- ... -->
```
This code not work, becose `SERVICE_PREFIX = 'Zend\\Navigation\\'`.
The code above can be correct if will be `SERVICE_PREFIX = 'ZendNavigation'`.
It is not most heavy problem.
I not understand why you are not understand me?

**Comment by froschdesign** [comment]
> This code not work, becose SERVICE_PREFIX = 'Zend\\Navigation\\'.

This is not true. The code works.
Again, I have tested the complete scenario with zend-mvc (3.0.4) and zend-navigation (2.8.1).

> The code above can be correct if will be SERVICE_PREFIX = 'ZendNavigation'.

No, because it will break all code with `zend-view`:

```
Fatal error: Uncaught Zend\ServiceManager\Exception\ServiceNotFoundException:
Unable to resolve service "Zend\Navigation\Default" to a factory;
are you certain you provided it during configuration?
```

The documentation is correct. The usage in Twig is different.

**Comment by novrm** [comment]
You are right.
I used the debug in controller. Both the cases work correctly.

```php
$renderer = $this->getServiceManager()->get('ViewRenderer');

$menu = $renderer->navigation('Zend\\Navigation\\layoutBodyHeaderBottomNavbarCollapseNav')->menu()->escapeLabels(false)->render();
\Zend\Debug\Debug::dump($menu);

$menu = $renderer->navigation('Zend\Navigation\layoutBodyHeaderBottomNavbarCollapseNav')->menu()->escapeLabels(false)->render();
 \Zend\Debug\Debug::dump($menu);
```
The usage in Twig is different.
But...
My view - Twig render a template as in 'use strict' mode...
A bit strange to write in framework `"Zend\\Navigation\\"` and apply in application `"Zend\Navigation\"`...

**Comment by froschdesign** [comment]
> It says to use <?= $this->navigation('default')->menu() ?>, but you will get an error then.

[Fixed with `zend-view` 2.9.0](https://github.com/zendframework/zend-view/releases/tag/release-2.9.0)
